### PR TITLE
Added dot net core support.

### PR DESCRIPTION
**Pull request by luwangshell**
New solution and project files for .net 5 are added to the repo.
Few updates to tweak the source code for .net core capabilities. For the ones which may not be compatible with both .net framework and .net core, added new files with Core suffix.

### CODE REVIEW

File: Rest/RestTests/XmlObjectTests.cs [hunk 1]
```diff
@@ -156,9 +157,13 @@ public void XmlObjectEvaluateTest()
...
-            Assert.AreEqual(items, evaluation);
-            Assert.AreEqual("3349", xmlObject.Evaluate("/data/item[@id='myId']"));
-            Assert.AreEqual("2268", xmlObject.Evaluate("/data/item[@id='otherId']"));
+
+            //Assert.AreEqual(items, evaluation);
+            Assert.AreEqual(0, string.Compare(items, evaluation, CultureInfo.CurrentCulture, CompareOptions.IgnoreSymbols));
+            //Assert.AreEqual("3349", xmlObject.Evaluate("/data/item[@id='myId']"));
+            Assert.AreEqual(0, string.Compare("3349", xmlObject.Evaluate("/data/item[@id='myId']"), CultureInfo.CurrentCulture, CompareOptions.IgnoreSymbols));
+            //Assert.AreEqual("2268", xmlObject.Evaluate("/data/item[@id='otherId']"));
+            Assert.AreEqual(0, string.Compare("2268", xmlObject.Evaluate("/data/item[@id='otherId']"), CultureInfo.CurrentCulture, CompareOptions.IgnoreSymbols));
```
Comment (essenius) [inline Rest/RestTests/XmlObjectTests.cs:164]: Why the IgnoreSymbols? Please add a comment

File: Rest/RestTests/RestRequestTests.cs [hunk 2]
```diff
@@ -40,9 +41,14 @@ public void RestRequestSetBodyForGetTest()
             var uri = new Uri("http://localhost/api");
             var factory = new RestRequestFactory();
             var restRequest = factory.Create(uri, context);
-            var target = new PrivateObject(restRequest);
-            target.Invoke("SetBody", "hello", Encoding.GetEncoding("iso-8859-1"), "GET");
-            var req = target.GetFieldOrProperty("_request") as HttpWebRequest;
+            // var target = new PrivateObject(restRequest);
```
Comment (essenius) [inline Rest/RestTests/RestRequestTests.cs]: Please remove all commented out code

**Review by essenius — CHANGES_REQUESTED** [review]
See comments in the code.

File: Rest/RestTests/XmlObjectTests.cs (hunk 1 above)
Comment (luwangshell) [reply Rest/RestTests/XmlObjectTests.cs:164]: The ignoresysmbols is used here for cross platform support. e.g. /r/n for windows, /n for Mac. Will add a inline comment here too.

**Comment by luwangshell** [comment]
Hi @essenius the recommended changes are pushed to this branch. Also I have added a workflow for .net core ci. Currently only one test is failing on the typicode because the cookie is no longer returned. The build on .net core CI: https://github.com/essenius/FitNesseFitSharpRest/pull/2/checks?check_run_id=3233928681#step:6:128, is it okay we remove the cookie assertion part for now?

**Comment by essenius** [comment]
@luwangshell OK to remove the cookie assertion

**Comment by luwangshell** [comment]
Now both workflow are running fine.

**Comment by luwangshell** [comment]
Have also updated the .net framework workflow to run all the unit tests as well.

**Comment by luwangshell** [comment]
Hi @essenius , now i have updated the project and source code to support both net 5 and net 48. Also updated the CI workflow to accommodate the changes: https://github.com/essenius/FitNesseFitSharpRest/actions/runs/1103896968

File: Rest/RestTests/RestTests.csproj [hunk 3]
```diff
@@ -1,163 +1,33 @@
...
-      <Project>{0BA5DBEB-3B3C-4BA7-B5FC-6719F23B3175}</Project>
-      <Name>Rest</Name>
-    </ProjectReference>
+    <ProjectReference Include="..\Rest\Rest.csproj" />
   </ItemGroup>
+
+ 
   <ItemGroup>
-    <Content Include="TestData.xml">
+    <None Update="TestData.xml">
```
Comment (essenius) [inline Rest/RestTests/RestTests.csproj]: If you moved the match parsing test data to code, this file should not be required anymore.

File: Rest/RestTests/SessionContextTests.cs [hunk 4]
```diff
@@ -37,27 +38,28 @@ public void SessionContextProxyTest()
         {
             var context = new SessionContext();
             // We could do this via SetDefaults, but that is a bit elaborate. So cheating with a PrivateObject
-            var target = new PrivateObject(context);
+            // var target = new PrivateObject(context);
```
Comment (essenius) [inline Rest/RestTests/SessionContextTests.cs]: Remove commented out code

File: Rest/Rest/Rest.csproj [hunk 5]
```diff
@@ -1,119 +1,21 @@
...
-    <Compile Include="Model\SessionContext.cs" />
-    <Compile Include="Utilities\FitNesseFormatter.cs" />
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="FitNesseRules.ruleset" />
-    <None Include="packages.config" />
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0'">
+	<PackageReference Include="Microsoft.AspNetCore.Http" Version="2.2.2" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Comment (essenius) [inline Rest/Rest/Rest.csproj]: Add Json to the framework independent item group (same for both)

File: .github/workflows/fitsharp-rest-ci.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        dotnet-version: 5.0.x
+        
+    - name: Restore dependencies
+      run: dotnet restore .\Rest\Rest.sln
+      
+    - name: Build
+      run: dotnet build .\Rest\Rest.sln
+      
+    - name: Test
+      run: dotnet test .\Rest\Rest.sln
```
Comment (essenius) [inline .github/workflows/fitsharp-rest-ci.yml]: add newline at end of file (multiple places)

File: Rest/RestTests/ExtensionFunctionsTests.cs [hunk 7]
```diff
@@ -41,6 +41,7 @@ public void ExtensionFunctionsIsLikeTest()
             Assert.IsTrue("abc".IsLike("a*"));
         }
 
+#if NET48
```
Comment (essenius) [inline Rest/RestTests/ExtensionFunctionsTests.cs]: We should be able to take the same approach for NET48 and NET50, and eliminate the need for TestData.xml

**Review by essenius — COMMENTED** [review]
Left a couple of comments in the code.

File: Rest/RestTests/SessionContextTests.cs (hunk 4 above)
Comment (luwangshell) [reply Rest/RestTests/SessionContextTests.cs]: fixed.

File: Rest/Rest/Rest.csproj (hunk 5 above)
Comment (luwangshell) [reply Rest/Rest/Rest.csproj]: fixed.

**Review by essenius — APPROVED** [review]

**Pull request merged**